src/tools/perf/lib/libperf_int.h

```diff
@@ -118,49 +122,75 @@ struct ucx_perf_thread_context {
     ucx_perf_result_t   result;
 };
 
-
 struct uct_peer {
     uct_ep_h                     ep;
     unsigned long                remote_addr;
     uct_rkey_bundle_t            rkey;
 };
 
-
 struct ucp_perf_request {
     void                         *context;
 };
 
+typedef struct {
+    ucs_status_t (*setup)(ucx_perf_context_t *perf);
+    void         (*cleanup)(ucx_perf_context_t *perf);
+    ucs_status_t (*run)(ucx_perf_context_t *perf);
+    void         (*barrier)(ucx_perf_context_t *perf);
+} ucx_perf_funcs_t;
 
-#define UCX_PERF_TEST_FOREACH(perf) \
-    while (!ucx_perf_context_done(perf))
-
-#define rte_call(_perf, _func, ...) \
-    ((_perf)->params.rte->_func((_perf)->params.rte_group, ## __VA_ARGS__))
-
+extern ucx_perf_funcs_t ucx_perf_funcs[];
 
 void ucx_perf_test_start_clock(ucx_perf_context_t *perf);
-
-
 void uct_perf_ep_flush_b(ucx_perf_context_t *perf, int peer_index);
-
-
 void uct_perf_iface_flush_b(ucx_perf_context_t *perf);
-
-
 ucs_status_t uct_perf_test_dispatch(ucx_perf_context_t *perf);
-
-
 ucs_status_t ucp_perf_test_dispatch(ucx_perf_context_t *perf);
-
-
 void ucx_perf_calc_result(ucx_perf_context_t *perf, ucx_perf_result_t *result);
+void uct_perf_barrier(ucx_perf_context_t *perf);
+void ucp_perf_barrier(ucx_perf_context_t *perf);
 
+ucs_status_t
+ucp_perf_test_alloc_host(const ucx_perf_context_t *perf, size_t length,
+                         void **address_p, ucp_mem_h *memh, int non_blk_flag);
+void ucp_perf_test_free_host(const ucx_perf_context_t *perf,
+                             void *address, ucp_mem_h memh);
+ucs_status_t ucp_perf_test_alloc_mem(ucx_perf_context_t *perf);
+void ucp_perf_test_free_mem(ucx_perf_context_t *perf);
+void ucx_perf_test_memcpy_host(void *dst, ucs_memory_type_t dst_mem_type,
+                               const void *src, ucs_memory_type_t src_mem_type,
+                               size_t count);
+ucs_status_t
+uct_perf_test_alloc_host(const ucx_perf_context_t *perf, size_t length,
+                         unsigned flags, uct_allocated_memory_t *alloc_mem);
+void uct_perf_test_free_host(const ucx_perf_context_t *perf,
+                             uct_allocated_memory_t *alloc_mem);
```

Comment:
move ucx_perf_global_init() to libperf_memory.c so these functions could be static and not have to be defined in H file